modules/client/src/test/kotlin/jp/co/soramitsu/iroha2/util/CryptoTest.kt

```diff
@@ -0,0 +1,99 @@
+package jp.co.soramitsu.iroha2.util
+
+import jp.co.soramitsu.iroha2.generateKeyPair
+import jp.co.soramitsu.iroha2.hash
+import jp.co.soramitsu.iroha2.hex
+import jp.co.soramitsu.iroha2.keyPairFromHex
+import jp.co.soramitsu.iroha2.sign
+import jp.co.soramitsu.iroha2.verify
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.async
+import kotlinx.coroutines.runBlocking
+import org.bouncycastle.util.encoders.Hex
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertDoesNotThrow
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+
+class CryptoTest {
+
+    @Test
+    fun `generating key pairs is thread safe`() {
+        val iterations = 1000
+        val futureResults = generateSequence { GlobalScope.async { generateKeyPair() } }
+            .take(iterations).toSet()
+
+        class ByteArrayWrapper(private val byteArray: ByteArray) {
+            override fun equals(other: Any?): Boolean {
+                if (this === other) return true
+                if (other !is ByteArrayWrapper) return false
+                if (!byteArray.contentEquals(other.byteArray)) return false
+                return true
+            }
+
+            override fun hashCode() = byteArray.contentHashCode()
+        }
+
+        runBlocking {
+            assertEquals(
+                iterations,
+                futureResults.map { it.await().private.encoded }
+                    .map { ByteArrayWrapper(it) }
+                    .toSet().size
+            )
+        }
+    }
+
+    @Test
+    fun `signature created and verified`() {
+        val keyPair = generateKeyPair()
+        val message = "Test message to sign.".toByteArray()
+        val signature = keyPair.private.sign(message)
+
+        assertTrue { keyPair.public.verify(signature, message) }
+    }
+
+    @Test
+    fun `hash function returns expected result`() {
+        val expected = "ba67336efd6a3df3a70eeb757860763036785c182ff4cf587541a0068d09f5b2"
+        val input = "6920616d2064617461"
+        assertEquals(expected, Hex.toHexString(input.hex().hash()))
+    }
+
+    @Test
+    fun `keypair serialized to hex and deserialized back`() {
+        val keyPair = generateKeyPair()
+        val pubKey = keyPair.public.encoded.hex()
+        val privKey = keyPair.private.encoded.hex()
+
+        val message = "foo".toByteArray()
+        val signature = keyPair.private.sign(message)
+
+        val restoredKeyPair = keyPairFromHex(pubKey, privKey)
+        assertEquals(keyPair.private, restoredKeyPair.private)
+        assertEquals(keyPair.public, restoredKeyPair.public)
+
+        assertDoesNotThrow {
+            assertTrue { restoredKeyPair.public.verify(signature, message) }
+        }
+    }
+
+    // @Test
+    // fun `transaction signature is same as expected`() {
```

Comment:
why is that commented?

Reply:
Yeah, need to fix this test. Signatures become not equal due random seed is generated every time. Will fix in the recent pr